[CI] OTLPMetricsIndexingRestIT testIngestMetricDataViaMetricExporter failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #10535 / debian-12_platform-support-unix](https://gradle-enterprise.elastic.co/s/24vvnanpprzcy)
- [elasticsearch-intake #28138 / part2](https://gradle-enterprise.elastic.co/s/33yeoil2klwf4)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:otel-data:javaRestTest" --tests "org.elasticsearch.xpack.oteldata.otlp.OTLPMetricsIndexingRestIT.testIngestMetricDataViaMetricExporter" -Dtests.seed=B3792081A893F6AE -Dtests.locale=ka-GE -Dtests.timezone=Pacific/Marquesas -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.oteldata.otlp.OTLPMetricsIndexingRestIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testIngestMetricDataViaMetricExporter'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: "2025-09-10T06:02:31Z"
     but: was "2025-09-10T06:02:31.000Z"
```

**Issue Reasons:**
- [main] 2 failures in test testIngestMetricDataViaMetricExporter (0.8% fail rate in 253 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -513,9 +513,6 @@ tests:
 - class: org.elasticsearch.xpack.logsdb.qa.BulkDynamicMappingChallengeRestIT
   method: testDateHistogramAggregation
   issue: https://github.com/elastic/elasticsearch/issues/134389
-- class: org.elasticsearch.xpack.oteldata.otlp.OTLPMetricsIndexingRestIT
-  method: testIngestMetricDataViaMetricExporter
-  issue: https://github.com/elastic/elasticsearch/issues/134426
 
 # Examples:
 #

--- a/x-pack/plugin/otel-data/src/javaRestTest/java/org/elasticsearch/xpack/oteldata/otlp/OTLPMetricsIndexingRestIT.java
+++ b/x-pack/plugin/otel-data/src/javaRestTest/java/org/elasticsearch/xpack/oteldata/otlp/OTLPMetricsIndexingRestIT.java
@@ -157,8 +157,8 @@ public class OTLPMetricsIndexingRestIT extends ESRestTestCase {
         ObjectPath search = search("metrics-generic.otel-default");
         assertThat(search.toString(), search.evaluate("hits.total.value"), equalTo(1));
         var source = search.evaluate("hits.hits.0._source");
-        assertThat(evaluate(source, "@timestamp"), equalTo(timestampAsString(now)));
-        assertThat(evaluate(source, "start_timestamp"), equalTo(timestampAsString(now)));
+        assertThat(Instant.parse(evaluate(source, "@timestamp")), equalTo(Instant.ofEpochMilli(TimeUnit.NANOSECONDS.toMillis(now))));
+        assertThat(Instant.parse(evaluate(source, "start_timestamp")), equalTo(Instant.ofEpochMilli(TimeUnit.NANOSECONDS.toMillis(now))));
         assertThat(evaluate(source, "_metric_names_hash"), isA(String.class));
         assertThat(ObjectPath.<Number>evaluate(source, "metrics.jvm\\.memory\\.total").longValue(), equalTo(totalMemory));
         assertThat(evaluate(source, "unit"), equalTo("By"));
@@ -411,10 +411,6 @@ public class OTLPMetricsIndexingRestIT extends ESRestTestCase {
         Map<String, Object> mapping = evaluate(mappings.values().iterator().next(), "mappings");
         assertThat(mapping, not(anEmptyMap()));
         return mapping;
-    }
-
-    private static String timestampAsString(long now) {
-        return Instant.ofEpochMilli(TimeUnit.NANOSECONDS.toMillis(now)).toString();
     }
 
     private void export(List<MetricData> metrics) throws Exception {

